### PR TITLE
Context rework

### PR DESCRIPTION
**Pull request by lgebhardt**
Reworks the context to make it part of a resource instance.
Removes resource from ValueFormatters to clarify purpose.
Renames object in resource to model.

### CODE REVIEW

File: lib/jsonapi/resource_controller.rb [hunk 1]
```diff
@@ -13,24 +13,23 @@ module JSONAPI
...
-          key_formatter: key_formatter
-        })
-        render_errors(@request.errors) unless @request.errors.empty?
-      rescue => e
-        handle_exceptions(e)
-      end
-    }
+    before_filter :setup_request
+
+    def setup_request
```
Comment (dgeb) [inline lib/jsonapi/resource_controller.rb]: probably should be a private method

File: README.md [hunk 2]
```diff
@@ -63,7 +63,7 @@ end
 
 This resource has 5 attributes: `:id`, `:name_first`, `:name_last`, `:email`, `:twitter`. By default these attributes must exist on the model that is handled by the resource.
 
-A resource object wraps a Ruby object, usually an ActiveModel record, which is available as the `@object` variable. This allows a resource's methods to access the underlying object.
+A resource object wraps a Ruby model, usually an ActiveModel record, which is available as the `@model` variable. This allows a resource's methods to access the underlying model.
```
Comment (dgeb) [inline README.md]: should still read "Ruby object" (not model)

**Comment by dgeb** [comment]
LGTM except those two minor issues :+1:

**Comment by lgebhardt** [comment]
Fixed. Thanks.

**Comment by dgeb** [comment]
Thanks @lgebhardt :shipit:

**Pull request merged**